src/app/components/category-products/category-products.component.ts

```diff
@@ -7,5 +18,122 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   encapsulation: ViewEncapsulation.Emulated,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class CategoryProductsComponent {
+export class CategoryProductsComponent implements AfterViewInit {
+  readonly categoryList$: Observable<CategoryModel[]> = this._categoriesService.getAllCategories();
+  readonly categoryId$: Observable<string> = this._activatedRoute.params.pipe(
+    map((params) => params['categoryId']),
+    shareReplay(1)
+  );
+  readonly queryParams$: Observable<QueryParamsQueryModel> = this._activatedRoute.queryParams.pipe(
+    map((params) => ({
+      sort: params['sort'] ?? 'featureValue',
+      order: params['order'] ?? 'desc'
+    })),
+    shareReplay(1),
+    tap((data) => this.setControlFromQueryParams(data))
+  );
+
+  readonly sortingOpts$: Observable<SortOptionModel[]> = of([
+    { display: 'Featured', key: 'featureValue;desc', order: 'desc' },
+    { display: 'Price Low to High', key: 'price;asc', order: 'asc' },
+    { display: 'Price High to Low', key: 'price;desc', order: 'desc' },
+    { display: 'Avg. Rating', key: 'ratingValue;desc', order: 'desc' }
+  ]);
+
+  readonly categoryDetails$: Observable<CategoryModel> = this.categoryId$.pipe(
+    switchMap((categoryId) => this._categoriesService.getSingleCategoryById(categoryId))
+  );
+  readonly productInCategoryList$: Observable<ProductQueryModel[]> = this.categoryId$.pipe(
+    switchMap((categoryId) =>
+      this._productsService
+        .getAllProducts()
+        .pipe(
+          map((products) =>
+            products
+              .filter((product) => product.categoryId === categoryId)
+              .map((product) => this.mapToProductQueryModel(product))
+          )
+        )
+    )
+  );
+  readonly sortForm: FormControl = new FormControl();
+
+  readonly productsFilteredAndSorted$: Observable<ProductQueryModel[]> = combineLatest([
+    this.queryParams$,
+    this.productInCategoryList$
+  ]).pipe(
+    map(([params, products]) =>
+      products.sort((a, b) => {
+        if (params.sort === 'featureValue') {
```

Comment:
too complex, you can do it without any if statement

Reply:
I managed to shorten this function to 2 ifs...
is that better?
`products.sort((a, b) => {
        const prev = Object.assign(a);
        const next = Object.assign(b);
        if (prev[params['sort']] > next[params['sort']]) {
          return params['order'] === 'asc' ? 1 : -1;
        }
        if (prev[params['sort']] < next[params['sort']]) {
          return params['order'] === 'asc' ? -1 : 1;
        } else return 0;
      })`